src/test/java/org/takes/misc/VerboseIteratorTest.java

```diff
@@ -0,0 +1,128 @@
+
+/**
+ * Tests for {@link VerboseIterator}.
+ * @author Marcus Sanchez (sanchez.marcus@gmail.com)
+ * @version $Id$
+ * @since 0.15.1
+ */
+public class VerboseIteratorTest {
+
+    /**
+     * VerboseIterator can return next value on a valid list.
+     */
+    @Test
+    public final void returnsNextValue() {
+        final String accept = "Accept: text/plain";
+        MatcherAssert.assertThat(
+            new VerboseIterable<String>(
+                Arrays.asList(
+                    accept,
+                    "Accept-Charset: utf-8",
+                    "Authorization: Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==",
+                    "Cache-Control: no-cache",
+                    "From: user@example.com"
+                ),
+                "Empty Error Message"
+            ).iterator().next(),
+            Matchers.equalTo(accept)
+        );
+    }
+
+    /**
+     * VerboseIterator can inform has a next value on a valid list.
+     */
+    @Test
+    public final void informsHasNextValue() {
+        MatcherAssert.assertThat(
+            new VerboseIterable<String>(
+                Arrays.asList(
+                    "User-Agent: LII-Cello/1.0 libwww/2.5",
+                    "accept-extension = \";\"",
+                    "Accept: text/*;",
+                    "Accept-Encoding: gzip;q=1.0"
+                ),
+                "this error should not be thrown"
+            ).iterator().hasNext(), Matchers.equalTo(true)
+        );
+    }
+
+    /**
+     * VerboseIterator next value throws exception on an empty list.
+     */
+    @Test(expected = RuntimeException.class)
+    public final void nextValueThrowsExceptionOnEmptyList() {
+        new VerboseIterable<String>(
+            Collections.<String>emptyList(),
+            "Expected Error Message"
+        ).iterator().next();
+    }
+
+    /**
+     * VerboseIterator returns false in has next value on empty list.
+     */
+    @Test
+    public final void returnFalseInHasNextValueOnEmptyList() {
+        MatcherAssert.assertThat(
+            new VerboseIterable<String>(
+                Collections.<String>emptyList(),
+                "Non used Error Message"
+            ).iterator().hasNext(),
+            Matchers.equalTo(false)
+        );
+    }
+
+    /**
+     * VerboseIterator can remove a value.
+     */
+    @Test(expected = UnsupportedOperationException.class)
+    public final void removeValue() {
+        final List<String> valid = Arrays.asList(
+            "If-None-Match: \"737060cd8c284d8af7ad3082f209582d\"",
+            "If-Range: \"737060cd8c284d8af7ad3082f209582d\"",
+            "Max-Forwards: 10",
+            "Origin: http://www.example-social-network.com",
+            "User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:12.0) Gecko/2010010"
+        );
+        final Iterable<String> verbose = new VerboseIterable<String>(
+            valid, "Thrown Error Message"
+        );
+        verbose.iterator().remove();
+        MatcherAssert.assertThat(
```

Comment:
I had mention the reason for having this extra code in a "todo" but was removed due to style checks.
Is the .remove functionality  going to be built? 
I found this error when creating this test case, but since I don't know if this is a problem or a design decision, I leave the extra code there.